src/main/java/com/ning/maven/plugins/dependencyversionscheck/AbstractDependencyVersionsMojo.java

```diff
@@ -487,23 +544,25 @@ private boolean isVisible(final String scope, final String [] validScopes)
      */
     private void addToResolutionMap(final Map resolutionMap, final VersionResolution resolution)
     {
-        List resolutions = (List) resolutionMap.get(resolution.getDependencyName());
-        if (resolutions == null) {
-            resolutions = new ArrayList();
-            resolutionMap.put(resolution.getDependencyName(), resolutions);
-        }
-
-        for (Iterator it = resolutions.iterator(); it.hasNext(); ) {
-            final VersionResolution existingResolution = (VersionResolution) it.next();
-            // TODO: It might be reasonable to fail the build in this case. However, I have yet to see
-            // this message... :-) 
-            if (!existingResolution.getActualVersion().equals(resolution.getActualVersion())) {
-                LOG.warn("Dependency '{} expects version '{}' but '{}' already resolved to '{}'!", 
-                         new Object [] { resolution.getDependencyName(), resolution.getActualVersion(), existingResolution.getDependencyName(), existingResolution.getActualVersion() });
+        synchronized (resolution.getDependencyName().intern()) {
```

Comment:
Note that `ConcurrentHashMap` in Java 8 has `computeIfAbsent` which has similar semantics (though the locks are larger, so Guava cache should be used if the compute method is not fast).